### PR TITLE
Repository for CategoryController, HomeController, RatingController, …

### PR DESCRIPTION
**Pull request by ghost**
…ShoppingController, ProfileController

## Related Tickets:
- [#Ticket ID](https://edu-redmine.sun-asterisk.vn/issues/26122)

## Notes Impacted Areas:
- N/A.

## Attachments:
- N/A.

### CODE REVIEW

**Comment by hoangminhuet** [comment]
ready

File: app/Http/Controllers/CategoryController.php [hunk 1]
```diff
@@ -2,97 +2,62 @@
...
-     * Store a newly created resource in storage.
-     *
-     * @param  \Illuminate\Http\Request  $request
-     * @return \Illuminate\Http\Response
-     */
     public function store(Request $request)
     {
-        Category::create($request->all());
+        $data = $request->all();
+        $this->categoryRepo->create($data);
```
Comment (ducls-2223) [inline app/Http/Controllers/CategoryController.php]: Có thể viết short code như sau: `$this->categoryRepo->create($request->all())`

File: app/Http/Controllers/CategoryController.php [hunk 2]
```diff
@@ -2,97 +2,62 @@
...
-     * @param  \Illuminate\Http\Request  $request
-     * @param  int  $id
-     * @return \Illuminate\Http\Response
-     */
     public function update(Request $request, $id)
     {
-        $category = Category::findOrFail($id);
-        $category->update($request->all());
+        $data = $request->all();
+        $this->categoryRepo->update($id, $data);
```
Comment (ducls-2223) [inline app/Http/Controllers/CategoryController.php]: tương tự

File: app/Http/Controllers/HomeController.php [hunk 3]
```diff
@@ -2,39 +2,52 @@
...
 class HomeController extends Controller
 {
+    protected $productRepo;
+    protected $ratingRepo;
+
+    public function __construct(
+        ProductRepositoryInterface $productRepo,
+        RatingRepositoryInterface $ratingRepo
+    )
+    {
```
Comment (huyenttm-0901) [inline app/Http/Controllers/HomeController.php]:
```suggestion
    ) {
```

File: app/Http/Controllers/ShoppingController.php [hunk 4]
```diff
@@ -2,58 +2,83 @@
...
+    protected $categoryRepo;
+    protected $sizeRepo;
+
+    public function __construct(
+        ProductRepositoryInterface $productRepo,
+        RatingRepositoryInterface $ratingRepo,
+        CategoryRepositoryInterface $categoryRepo,
+        SizeRepositoryInterface $sizeRepo
+    )
+    {
```
Comment (huyenttm-0901) [inline app/Http/Controllers/ShoppingController.php]: tương tự

**Comment by hoangminhuet** [comment]
ready

**Pull request merged**